mne/io/nsx/tests/test_nsx.py

```diff
@@ -0,0 +1,261 @@
+from mne.datasets.testing import data_path, requires_testing_data
+from mne.io.tests.test_raw import _test_raw_reader
+from mne.io.constants import FIFF
+from mne import make_fixed_length_epochs
+
+
+testing_path = data_path(download=False)
+nsx_21_fname = os.path.join(testing_path, "nsx", "test_NEURALSG_raw.ns3")
+nsx_22_fname = os.path.join(testing_path, "nsx", "test_NEURALCD_raw.ns3")
+nsx_31_fname = os.path.join(testing_path, "nsx", "test_BRSMPGRP_raw.ns3")
+nsx_test_fname = os.path.join(testing_path, "nsx", "Test_anonymized.ns3")
+
+
+def test_decode_online_filters():
+    """Tests for online low/high-pass filter decoding."""
+    info = _empty_info(100.0)
+    highpass = np.array([0.0, 0.1])
+    lowpass = np.array([50, 50])
+    with pytest.raises(RuntimeWarning, match="different highpass filters"):
+        _decode_online_filters(info, highpass, lowpass)
+    assert info["highpass"] == 0.1
+
+    info = _empty_info(100.0)
+    highpass = np.array([0.0, 0.0])
+    lowpass = np.array([40, 50])
+    with pytest.raises(RuntimeWarning, match="different lowpass filters"):
+        _decode_online_filters(info, highpass, lowpass)
+    assert info["lowpass"] == 40
+
+    info = _empty_info(100.0)
+    highpass = np.array(["NaN", "NaN"])
+    lowpass = np.array(["NaN", "NaN"])
+    _decode_online_filters(info, highpass, lowpass)
+    assert info["highpass"] == 0.0
+    assert info["lowpass"] == 50.0
+
+
+@requires_testing_data
+def test_nsx_ver_31():
+    """Primary tests for BRSMPGRP reader."""
+    raw = read_raw_nsx(nsx_31_fname)
+    assert getattr(raw, "_data", False) is False
+    assert raw.info["sfreq"] == 2000
+
+    # Check info object
+    assert raw.info["meas_date"].day == 31
+    assert raw.info["meas_date"].year == 2023
+    assert raw.info["meas_date"].month == 1
+    assert raw.info["chs"][0]["cal"] == 0.6103515625
+    assert raw.info["chs"][0]["range"] == 0.001
+
+    # Check raw_extras
+    for r in raw._raw_extras:
+        assert r["orig_format"] == raw.orig_format
+        assert r["orig_nchan"] == 128
+        assert len(r["timestamp"]) == len(r["nb_data_points"])
+        assert len(r["timestamp"]) == len(r["offset_to_data_block"])
+
+    # Check annotations
+    assert raw.annotations[0]["onset"] * raw.info["sfreq"] == 101
+    assert raw.annotations[0]["duration"] * raw.info["sfreq"] == 49
+
+    # Ignore following RuntimeWarning in mne/io/base.py in _write_raw_fid
+    # "Acquisition skips detected but did not fit evenly into output"
+    # "buffer_size, will be written as zeroes."
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", RuntimeWarning)
+        raw = _test_raw_reader(
+            read_raw_nsx,
+            input_fname=nsx_31_fname,
+            eog=None,
+            misc=None,
+            test_scaling=False,
+            test_rank=False,
+        )
+    raw_data, times = raw[:]
+    n_channels, n_times = raw_data.shape
+    assert times.shape[0] == n_times
```

Comment:
I'm not sure this is the most useful test, now that `raw` is a `RawNSX` these kinds of round trips are already tested. I would hardcode values in for the number of times that you know to be true from the data file and a checksum on the values would be nice too.